src/rust/engine/process_execution/src/cache_tests.rs

```diff
@@ -98,6 +99,7 @@ fn run_roundtrip(script_exit_code: i8) -> RoundtripResults {
 }
 
 #[test]
+#[ignore] //TODO(gshuflin) this test will fail on a mac until a follow-up removes the hard-coding of "Linux" in the cache.
```

Comment:
If you default your placeholder to `Platform::current().unwrap()` rather than `Platform::Linux`, this test will keep working in the mean time, while being equally wrong in production :)

Reply:
@illicitonion doing that breaks a bunch of other tests; cf. the below comment I'm just gonna rebase the PR that actually finishes implementing the feature properly on top of these commits, even if it makes for a long PR.